Formula/v8.rb

```diff
@@ -15,98 +17,79 @@ class V8 < Formula
-  end
-
-  resource "gtest" do
-    url "https://chromium.googlesource.com/external/github.com/google/googletest.git",
-        :revision => "6f8a66431cb592dad629028a50b3dd418a408c87"
-  end
-
-  resource "gmock" do
-    url "https://chromium.googlesource.com/external/googlemock.git",
-        :revision => "0421b6f358139f02e102c9c332ce19a33faf75be"
-  end
-
-  resource "clang" do
-    url "https://chromium.googlesource.com/chromium/src/tools/clang.git",
-        :revision => "faee82e064e04e5cbf60cc7327e7a81d2a4557ad"
-  end
-
   def install
-    # Bully GYP into correctly linking with c++11
-    ENV.cxx11
-    ENV["GYP_DEFINES"] = "clang=1 mac_deployment_target=#{MacOS.version}"
-    # https://code.google.com/p/v8/issues/detail?id=4511#c3
-    ENV.append "GYP_DEFINES", "v8_use_external_startup_data=0"
-
-    # fix up libv8.dylib install_name
-    # https://github.com/Homebrew/homebrew/issues/36571
-    # https://code.google.com/p/v8/issues/detail?id=3871
-    inreplace "tools/gyp/v8.gyp",
-              "'OTHER_LDFLAGS': ['-dynamiclib', '-all_load']",
-              "\\0, 'DYLIB_INSTALL_NAME_BASE': '#{opt_lib}'"
-
-    (buildpath/"build/gyp").install resource("gyp")
-    (buildpath/"third_party/icu").install resource("icu")
-    (buildpath/"buildtools").install resource("buildtools")
-    (buildpath/"base/trace_event/common").install resource("common")
-    (buildpath/"tools/swarming_client").install resource("swarming_client")
-    (buildpath/"testing/gtest").install resource("gtest")
-    (buildpath/"testing/gmock").install resource("gmock")
-    (buildpath/"tools/clang").install resource("clang")
-
-    system "make", "native", "library=shared", "snapshot=on",
-                   "console=readline", "i18nsupport=off",
-                   "strictaliasing=off"
-
-    include.install Dir["include/*"]
-
-    cd "out/native" do
-      rm ["libgmock.a", "libgtest.a"]
-      lib.install Dir["lib*"]
-      bin.install "d8", "mksnapshot", "process", "shell" => "v8"
+    ENV.prepend_path "PATH", buildpath
+
+    v8_version = head? ? "HEAD" : version
+    arch = MacOS.prefer_64_bit? ? "x64" : "x86"
+    output_name = "#{arch}.release"
+    output_path = "out.gn/#{output_name}"
+
+    # This env variable used by gclient to prevent depot_tools to update depot_tools on every call
+    # see https://www.chromium.org/developers/how-tos/depottools#TOC-Disabling-auto-update
+    ENV["DEPOT_TOOLS_UPDATE"] = "0"
+
+    # Initialize and sync gclient
+    system "gclient", "root"
+    system "gclient", "config", "--spec", <<~EOS
+      solutions = [
+        {
+          "url": "https://chromium.googlesource.com/v8/v8.git",
+          "managed": False,
+          "name": "v8",
+          "deps_file": "DEPS",
+          "custom_deps": {},
+        },
+      ]
+      target_os = [ "mac" ]
+      target_os_only = True
+    EOS
+    system "gclient", "sync", "-vvv", "-j", Hardware::CPU.cores, "-r", v8_version
```

Comment:
You could add `--no-history` here to only make a shallow clone, because we won`t reusing the repos anyway.